### ISSUE TITLE
fixup messages should not trigger a gitlint violation

### ISSUE THREAD

**Issue by nud** [jorisroovers/gitlint#33]
When using `git commit --fixup=<commitid>`, some gitlint violations are triggered:

```
$ git commit -a --fixup=HEAD
gitlint: checking commit message...
3: B6 Body message is missing
-----------------------------------------------
gitlint: Your commit message contains the above violations.
Continue with commit anyways (this keeps the current commit message)? [y(es)/n(no)/e(dit)] 
```

Probably the B6 check should ignore commits whose title starts with `fixup!` ?

**Comment by jorisroovers** [comment]
Ok, I have never used ```--fixup``` or ```--squash``` before in combination with ```commit```, I always take care of that manually during rebasing. TIL :)

For reference for others, here's some info: https://robots.thoughtbot.com/autosquashing-git-commits

I agree this is valid. Will be fixed, but as often, it might take me a while to get to this. Happy to accept PRs :)

Thanks for submitting!

**Comment by nud** [comment]
Thanks a lot!
